bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerCreateOp.java

```diff
@@ -125,28 +125,52 @@
      * Initiates the operation.
      */
     public void initiate() {
+        int actualEnsembleSize = ensembleSize;
+        List<BookieId> ensemble = null;
+        // select bookies for first ensemble
+        if (bk.getConf().getOpportunisticStriping()) {
+            BKNotEnoughBookiesException lastError = null;
+            // we would like to select ensembleSize bookies, but
+            // we can settle to writeQuorumSize
+            while (actualEnsembleSize >= writeQuorumSize) {
+                try {
+                    ensemble = bk.getBookieWatcher()
+                        .newEnsemble(actualEnsembleSize, writeQuorumSize, ackQuorumSize, customMetadata);
+                    lastError = null;
+                    break;
+                } catch (BKNotEnoughBookiesException e) {
+                    LOG.error("Not enough bookies to create ledger with {}/{}/{}",
+                            actualEnsembleSize, writeQuorumSize, ackQuorumSize);
+                    lastError = e;
+                    actualEnsembleSize--;
+                }
+            }
+            if (lastError != null) {
+                createComplete(lastError.getCode(), null);
+                return;
+            }
+        } else {
+            try {
+                ensemble = bk.getBookieWatcher()
+                        .newEnsemble(actualEnsembleSize, writeQuorumSize, ackQuorumSize, customMetadata);
+            } catch (BKNotEnoughBookiesException e) {
+                LOG.error("Not enough bookies to create ledger with {}/{}/{}",
```

Comment:
Very useful error log, can we also have some details as to what the three arguments are?
something like `Not enough bookies to create ledger with EnsembleSize/WriteQuorum/AckQuorum : {}/{}/{}`
So it'll be helpful for users to know which is which.